### DISCUSSION TITLE
Preview not picking up updated CSS

### DISCUSSION THREAD

**Discussion by draiggoch-keith** [phcode-dev/phoenix#1991] [Q&A]
When I make a change in my CSS file the change does not show in the Live Preview.  If I "Inspect" it still shows the old version of the styles.  Same problem going to  Chrome or Firefox from the Live Preview buttons.  Ctrl-F5 does not help.  The only way to see the styles change is to exit and restart Phoenix. Changes in the html DO show OK in the Live Preview.

Also on my Win 10 PC I get "Error opening live preview in chrome".

I'm running PCs with both Windows 10 and 11 Home,  XAMPP, Chrome and Firefox.  All fully updated.  I have reinstalled XAMPP and Phoenix, and tried using ports 80 and 8080 - all no differences.

If I just run localhost:8080/myproject/index.html in a browser it works fine - Ctrl-F5 will load the updated CSS file, but that's nothing like as convenient for developing.

I've been using Brackets for several years and it still does the Live Preview fine, without these problems.

Any suggestions what to try please - I've wasted a ton of time trying to fix this?

**Comment by abose** [comment]
Can you open the default project from `file menu - start project` and see if live preview works when editing css in default project?

**Comment by draiggoch-keith** [comment]
Live preview works fine with the default project.  So I am now actively trying to track down exactly what difference between the default project and mine caused the problem.  All along it's seemed like there is a cache somewhere that is not getting refreshed when needed.

**Reply by abose** [reply]
If you use any less or scss files that may be an issue. You are not using any custom server for the project right? https://docs.phcode.dev/docs/Features/live-preview-settings. Custom servers only support css reload and not live previews

**Comment by draiggoch-keith** [comment]
I'm not using less or scss.  But I AM using a custom server.  I'm doing that because I have include files and without the custom server they don't get brought in. I've tried with a .HTML project and a .PHP project.  I am seeing a situation where the in-app preview window shows a stale css yet the pop-out (which opens in Chrome) shows the updated css correctly.  (I accept I need to do a refresh.) It's all very quirky - I'm still trying to find a simple case to narrow down exactly what the trigger is between working and not.

**Reply by abose** [reply]
What is the server you are using. step by step instructions on how to reproduce this issue will help be debug this issue.

**Comment by draiggoch-keith** [comment]
I'm using XAMPP for Windows v8.2.12.  I've been trying very hard to produce a minimal repeatable case, but without success because I cannot see any pattern for when things work and do not.  Here is some of what I have found - maybe it will help you!

1. Problems only seem to happen when I use a Custom Development Server.  I do this because both my HTML and PHP projects use include files. Without this server the include files in the HTML do not get brought in.  PHP always requires the server.
2. I have created a new folder with new code and everything works fine - even with the custom server the style changes in the css showed in the preview.
3. I have then copied those files to a folder that had previously been used for a project with Phoenix, but completely emptied.  Everything should have been 100% the same, but now stale css styles were being picked up (the original problem).
4. In that same folder I alternated between  using the custom server and not using it.  The only difference should have been that the include file text appeared and disappeared - but in fact the styles changed.
5. I still get the message "Error Opening Live Preview in chrome" when clicking the tiny Chrome icon.  But if I click the "popout" button the preview does open in Chrome.

I wish I could come up with something repeatable as without that I can't expect you to find the solution - maybe my rough  hints above with help a bit.   If I can come up with anything else or better I'll pas it on.  Thanks for you efforts (and for all the fine coding that's in Phoenix).

**Reply by abose** [reply]
Thanks for the detailed notes. Point 5 is unrelated but all others looks like some kind of caching issue either at the browser or at server. I tested with some smaple projects with custom dev servers and found things to be working, but let me check a bit more.
